File: go/test/protocol/test_case_received_telemetry.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type TestCaseCloudEventCapsule struct {
	CloudEvent *TestCaseCloudEvent `yaml:"cloud-event"`
}

type testCaseReceivedTelemetry struct {
	TelemetryValue any                        `yaml:"telemetry-value"`
	Metadata       *map[string]string         `yaml:"metadata"`
	TopicTokens    *map[string]string         `yaml:"topic-tokens"`
	Capsule        *TestCaseCloudEventCapsule `yaml:",inline"`
	SourceIndex    *int                       `yaml:"source-index"`
}

type TestCaseReceivedTelemetry struct {
	testCaseReceivedTelemetry
}

func (receivedTelemetry *TestCaseReceivedTelemetry) UnmarshalYAML(
	node *yaml.Node,
) error {
	*receivedTelemetry = TestCaseReceivedTelemetry{}

	receivedTelemetry.TelemetryValue = false

	return node.Decode(&receivedTelemetry.testCaseReceivedTelemetry)
}

File: rust/azure_iot_operations_protocol/tests/metl/test_case_received_telemetry.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use std::collections::HashMap;

use serde::Deserialize;

use crate::metl::optional_field::deserialize_optional_field;
use crate::metl::test_case_cloud_event::TestCaseCloudEvent;

#[derive(Clone, Deserialize, Debug)]
#[allow(dead_code)]
pub struct TestCaseReceivedTelemetry {
    #[serde(rename = "telemetry-value")]
    #[serde(default)]
    #[serde(deserialize_with = "deserialize_optional_field")]
    #[allow(clippy::option_option)]
    pub telemetry_value: Option<Option<String>>,

    #[serde(rename = "metadata")]
    #[serde(default)]
    pub metadata: HashMap<String, Option<String>>,

    #[serde(rename = "topic-tokens")]
    #[serde(default)]
    pub topic_tokens: HashMap<String, String>,

    #[serde(rename = "cloud-event")]
    #[serde(default)]
    #[serde(deserialize_with = "deserialize_optional_field")]
    #[allow(clippy::option_option)]
    pub cloud_event: Option<Option<TestCaseCloudEvent>>,

    #[serde(rename = "source-index")]
    pub source_index: Option<i32>,
}
